src/codeu/chat/server/LocalFile.java

```diff
@@ -0,0 +1,299 @@
+
+    public static final String MESSAGE_FILE_NAME = "/msgDat.sav";
+    public static final String USER_FILE_NAME = "/usrDat.sav";
+    public static final String CONVERSATION_FILE_NAME = "/cvrsDat.sav";
+
+    //Instance varibles for saving the current data of server.
+    private final LinkedHashSet<User> users;
+    private final LinkedHashSet<ConversationHeader> conversationHeaders;
+    private final LinkedHashSet<Message> messages;
+
+    private final File file;
+
+    private boolean hasUserModified = false;//It indicates if there is a new data should be handled.
+    private boolean hasMessageModified = false;
+    private boolean hasConversationModified = false;
+
+    //false: The instance is not initialized. All new data will not mark the status as modified.
+    //true: All new data been added to this instance should mark the status as modified.
+    private boolean isInitialized = false;
+
+    private final Serializer<Collection<Message>> localMessages = Serializers.collection(Message.SERIALIZER);
+    private final Serializer<Collection<ConversationHeader>> localConversationHeaders = Serializers.collection(ConversationHeader.SERIALIZER);
+    private final Serializer<Collection<User>> localUsers = Serializers.collection(User.SERIALIZER);
+
+    private final File userFile;
+    private final File conversationFile;
+    private final File messageFile;
+    public LocalFile (File file)
+    {
+        this.file = file;
+        users = new LinkedHashSet<>();
+        conversationHeaders= new LinkedHashSet<>();
+        messages= new LinkedHashSet<>();
+
+        userFile = new File(file.getPath() + USER_FILE_NAME);
+        conversationFile = new File(file.getPath() + CONVERSATION_FILE_NAME);
+        messageFile = new File(file.getPath() + MESSAGE_FILE_NAME);
+        try
+        {
+            if(!userFile.exists())
+            {
+                userFile.createNewFile();
+            }      
+            if(!conversationFile.exists())
+            {
+                conversationFile.createNewFile();
+            }
+            if(!messageFile.exists())
+            {
+                messageFile.createNewFile();
+            }
+        }
+        catch(IOException exception)
+        {
+            LOG.error("Failed to create new file!");
+            exception.printStackTrace();
+            throw new RuntimeException();
+        }
+    }
+
+    /**
+     * Get a copy of users
+     * 
+     * @return  LinkedHashSet<User> Current users from this instance
+     */
+    public LinkedHashSet<User> getCopyOfUsers()
+    {
+        hasUserModified = false;
+        return new LinkedHashSet<>(users);
+    }
+    /**
+     * Get a copy of conversations
+     * 
+     * @return  LinkedHashSet<ConversationHeader> Current conversations from this instance
+     */
+    public LinkedHashSet<ConversationHeader> getCopyOfConversationHeaders()
+    {
+        hasConversationModified = false;
```

Comment:
This seems odd to me. Normally, a "getter" method shouldn't modify any state, and this would be unexpected behavior to anyone working with this class in the future. You should remove this line that sets hasConversationModified = false.